test/tests/serial/mail_with_ldap.bats

```diff
@@ -158,6 +158,53 @@ function teardown_file() {
   done
 }
 
+
+@test "dovecot: ldap email listing works" {
+  _run_in_container doveadm user '*'
+  assert_success
+  assert_line --index 0 'some.user@localhost.localdomain'
+  assert_line --index 1 'some.other.user@localhost.otherdomain'
+  assert_line --index 2 'some.user.email@localhost.localdomain'
+  _should_output_number_of_lines 3
+}
+
+@test "dovecot: (ENV DOVECOT_ITERATE_ATTRS=uniqueIdentifier=user) ldap email listing works" {
+  local CUSTOM_SETUP_ARGUMENTS=(
+    --env DOVECOT_ITERATE_ATTRS='uniqueIdentifier=user'
+  )
+
+  _run_in_container doveadm user '*'
+  assert_success
+  assert_line --index 0 'some.user'
+  assert_line --index 1 'some.other.user'
+  assert_line --index 2 'some.user.id'
+  _should_output_number_of_lines 3
+}
+
+@test "dovecot: (ENV DOVECOT_ITERATE_FILTER=(objectClass=PostfixBookMailAccount)) ldap email listing works" {
+  local CUSTOM_SETUP_ARGUMENTS=(
+    --env DOVECOT_ITERATE_FILTER='(objectClass=PostfixBookMailAccount)'
+  )
+
+  _run_in_container doveadm user '*'
+  assert_success
+  assert_line --index 0 'some.user@localhost.localdomain'
+  assert_line --index 1 'some.other.user@localhost.otherdomain'
+  assert_line --index 2 'some.user.email@localhost.localdomain'
+  _should_output_number_of_lines 3
+}
```

Comment:
Here's an example of what this would look like from the advice given in my review feedback:

```bash
@test "dovecot: (ENV DOVECOT_ITERATE_FILTER=(objectClass=PostfixBookMailAccount)) ldap email listing works" {
  # Container setup:
  export CONTAINER_NAME='dms-test_ldap_dovecot-iter3'
  local CONTAINER_ARGS_ENV_CUSTOM=(
    "${CONFIG_CONTAINER_BASE[@]}"
    --env DOVECOT_ITERATE_FILTER='(objectClass=PostfixBookMailAccount)'
  )
  _init_with_defaults
  _common_container_setup 'CONTAINER_ARGS_ENV_CUSTOM'

  # Actual test-case:
  _run_in_container doveadm user '*'
  assert_success
  assert_line --index 0 'some.user@localhost.localdomain'
  assert_line --index 1 'some.other.user@localhost.otherdomain'
  assert_line --index 2 'some.user.email@localhost.localdomain'
  _should_output_number_of_lines 3

  # Cleanup, removes test-case container:
  _default_teardown
}
```

The cleanup is just force removing the container:

https://github.com/docker-mailserver/docker-mailserver/blob/e9f04cf8a7d31efbc8ae9a4a35603409f637f3b4/test/helper/setup.bash#L198-L204

Setup is like we do in `setup()`, except we need to export the custom args as a common base var to expand / import here. It's important to give each container a unique name to identify it, I chose to use the name variant `dovecot-iter3` as it's the 3rd test-case container for testing against that ENV (_or 2nd if we don't create a separate container for the default test_).

Setting the name will ensure the helper methods used will know which container to use implicitly.